File: 4/4.py
import math
s=[0,0,0,0,0,0,0]
m=0
n=0
max=1
for i in range(999,101,-1):
    for j in range(999,101,-1):
        Mu=i*j
        k=10
        while Mu>=1:
            s[m]=Mu%10
            Mu=Mu/10
            m=m+1
        k=m-1
        m=0
        n=0
        Mu=i*j
        while s[n]==s[k] and k>=n:
           k=k-1
           n=n+1
           if k==n or k<n:
               if Mu>max:
                   max=Mu
print max

File: 4/4.c
#include<stdio.h>
#include<math.h>
int main()
{
    int i,j,m=0,max=0,n=0,multi;
    int a[6];
    for (i=101;i<1000;i++)
        for (j=101;j<1000;j++)
        {
            multi=i*j;
            int k=10;
            for(;multi>1;m++)
            {
                a[m]=multi%10;
                k=(int)pow(10,m+1);
                multi=(int)(multi/k);
                printf("%d",k);
            }
            multi=i*j;
            for(;(a[n]==a[m])&&(m>=n);m--,n++)
            {
                if (multi>max)
                {
                    max=multi;
                }
            }
             
        }
    printf("%d\n",max);
}
